src/util/ConfigManager/ConfigManager.h

```diff
@@ -46,6 +46,32 @@ concept Validator =
     std::regular_invocable<Func, const ParameterTypes&...> &&
     std::same_as<std::invoke_result_t<Func, const ParameterTypes&...>, bool>;
 
+// Simple struct, that holds an error message. For use as the return type of
+// invocable object, that fullfill `ExceptionValidator`.
+struct ErrorMessage {
+  const std::string message_;
```

Comment:
but why is it const?
In this case you should make the member `private` and only allow access to a `const&`, that is the pattern here.

Reply:
> but why is it const?

Because a `ErrorMessage` is only ever read after construction, existing as a data container for our validators.

Reply:
The advantage would be to still have the efficient move semantics which can move the string, but all other accesses are const.